### PR TITLE
[docs] Update docstring for repo_id in push_to_hub

### PR DESCRIPTION
**Pull request by tomaarsen**
Hello!

## Pull Request overview
* Update docstring for `repo_id` in `push_to_hub`.

## Details
This PR speaks for itself - the old text was a bit nonsensical. I think the new one is an improvement, but I'm open to alternatives, too.

- Tom Aarsen

### CODE REVIEW

File: src/huggingface_hub/hub_mixin.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def push_to_hub(
 
         Args:
             repo_id (`str`):
-                Repository name to which push.
+                Repository name to push to.
```
Comment (Wauplin) [inline src/huggingface_hub/hub_mixin.py:263]:
```suggestion
                ID of the repository to push to (example: `"username/my-model"`).
```

**Review by Wauplin — APPROVED** [review]
Thanks for reporting this and starting a PR @tomaarsen 
If you don't mind, I added suggestions to add an example to the description.

**Comment by tomaarsen** [comment]
Even better! I'm using
```python
            repo_id (`str`):
                The full repository name to push to, e.g. `"tomaarsen/setfit_sst2"`.
```

for my changes to SetFit to re-add support for `push_to_hub`.
Is there a strict definition of repository name vs ID? If so, then I can use that same verbiage in SetFit.

- Tom Aarsen

**Comment by tomaarsen** [comment]
That clears it up! Thanks. 

The test failures are surely unrelated - let me know if you need anything else from me for this PR :)

- Tom Aarsen

**Comment by Wauplin** [comment]
Nothing else needed! Thanks for the PR, I'm merging it :)

**Pull request merged**